docs/fee_design.md

```diff
@@ -0,0 +1,169 @@
+# Fee Exit Design
+
+This document describes the design for fee exits. It starts with the requirements, and then describes the basic fee mechanism within our Plasma M(ore)VP design. 
+
+## Requirement
+
+### Functional Requirement
+
+1. The operator can exit fees to an address the operator owns.
+2. Able to support fee exit for multiple transaction types where each transaction type can have different fee rules.
+3. Support the initial fee rule for Payment transaction, which is a fixed amount in cents per transaction.
+4. More fee rules could be added later. Here are some examples:
+   1. as a percentage of ETH gas. 
+   2. as a percentage of notional (aka transaction dollar amount).
+   3. fixed token price... floating with USD or other fiat.
+
+### Non-functional Requirement
+1. A Fee exit does not take longer time than normal exit.
+2. A Fee exit can batch-exit multiple fees collected in different transactions.
+
+### Out of Scope
+1. The fee rules are enforced by the operator by accepting transactions in the Child Chain, and not at the smart contracts level. A POA Plasma network always gives the operator the chance to censor transactions, therefore having the fee rule enforcement implemented in the smart contracts is not a design goal. However, users are protected by the watcher which would report a misbehaving operator if they tried to exit more fees than accepted in the recorded transactions. 
+
+## General fee mechanism design
+
+### High level description
+
+A Child Chain operator decides and enforces the fee format and the rule for transactions. The smart contracts do not check the rules for fee transactions. The rule enforcement is perfomed by the Child chain service. This is the case as long as the Plasma network runs in POA.
+
+The transaction fee is implied for a Payment transaction. It is the difference of the sum of inputs and the sum of outputs. For instance, if the sum of inputs is 10 ETH and sum of outputs is 9.9 Eth, the 0.1 Eth difference would be the transaction fee.
+
+In order to exit fees, an operator would first put a special fee transaction into a plasma chain block. After that, the operator can spend that fee transaction output to a payment transaction. So the operator can exit it as a normal payment transaction. This removes the need to deploy an Exit Game contract for the fee transaction type as it requires only an empty contract to represent the existence of the transaction type and its protocol (using MoreVP instead of MVP).
+
+A fee transaction is special and different from a normal transaction in that:
+
+1. It does not need to consume any inputs. As the fee is implied (at least for the Payment transactions), there is no output that consumes an input.
+2. The verification of the fee transaction relies on Child Chain and watcher only. 
```

Comment:
```suggestion
2. The verification of the fee transaction relies on Child Chain and Watcher only. 
```

I see this is re-occurring, sometimes we capitalize services sometimes not.

Reply:
Let's go with `Child Chain` and `Watcher`, will update everywhere.